composer/datasets/streaming/world.py

```diff
@@ -0,0 +1,82 @@
+from torch.utils.data import get_worker_info
+
+from composer.utils import dist
+
+
+class World(object):
+    """Context about workers, devices, and nodes. Useful for dividing work.
+
+    Fields:
+      - node / num_nodes
+
+      - global_device / global_num_devices
+      - device_of_node / devices_per_node
+
+      - global_worker / global_num_workers
+      - worker_of_node / workers_per_node
+      - worker_of_device / workers_per_device
+    """
+
+    def __init__(self, global_worker: int, num_nodes: int, devices_per_node: int, workers_per_device: int) -> None:
```

Comment:
I think there's a tradeoff of consistency vs. clarity here, it's a bit more readable to use fields like `global_device` and `num_global_devices` rather than `rank` and `world_size` which it's the user has to already know refer to devices rather than workers? Could we re-open this when we potentially refactor `world.py` into `dist.py`?